rust/core-lib/src/view.rs

```diff
@@ -449,26 +454,24 @@ impl View {
         self.wrap_col = wrap_col;
     }
 
-    pub fn after_edit(&mut self, text: &Rope, delta: &Delta<RopeInfo>, pristine: bool) {
-        let (iv, new_len) = delta.summary();
-        // Note: this logic almost replaces setting the cursor in Editor::commit_delta,
-        // but doesn't set col or scroll to the cursor. It could be extended to subsume
-        // that entirely.
-        // Also note: for committing plugin edits, we probably want to know the priority
-        // of the delta so we can set the cursor before or after the edit, as needed.
-        if self.sel_end >= iv.start() {
-            if self.sel_end >= iv.end() {
-                self.sel_end = self.sel_end - iv.size() + new_len;
-            } else {
-                self.sel_end = iv.start() + new_len;
-            }
-        }
-        self.sel_start = self.sel_end;
+    /// Updates the view after the text has been modified by the given `delta`.
+    /// This method is responsible for updating the cursors, and also for
+    /// recomputing line wraps.
+    ///
+    /// Return value is a location of a point that should be scrolled into view.
+    pub fn after_edit(&mut self, text: &Rope, delta: &Delta<RopeInfo>, pristine: bool)
+        -> Option<usize>
+    {
         if self.breaks.is_some() {
```

Comment:
`if let Some(breaks) = self.breaks.as_mut()`

and then modify the re-wrap line to use `breaks` instead of an unwrap.